Swiftification/SequenceType.swift

```diff
@@ -26,20 +26,33 @@ public extension SequenceType {
     
     /// Returns the first element of `self` that tests `true`, or `nil` if no element tests `true`.
     @warn_unused_result
-    func takeFirst(@noescape test: Generator.Element throws -> Bool) rethrows -> Generator.Element? {
+    func takeFirst(@noescape test: (Generator.Element) throws -> Bool) rethrows -> Generator.Element? {
         for element in self where try test(element) {
             return element
         }
         return nil
     }
     
+    /// Returns all elements until the first element of `self` that tests `false`.
+    @warn_unused_result
+    func takeWhile(@noescape test: (Generator.Element) throws -> Bool) rethrows -> [Generator.Element] {
+        var results: [Generator.Element] = []
+        for element in self {
+            if try !test(element) {
+                break
+            }
+            results.append(element)
+        }
+        return results
+    }
+    
     /// Returns the array of elements for which condition(element) is unique
     @warn_unused_result
-    func uniqueBy<T: Hashable>(@noescape condition: (Generator.Element) -> T) -> [Generator.Element] {
+    func uniqueBy<T: Hashable>(@noescape condition: (Generator.Element) throws -> T) rethrows -> [Generator.Element] {
```

Comment:
Won't it kind of stink that we'll have to surround this with a try/catch to use this now?